LAKMobile/src/context/AuthContext.tsx

```diff
@@ -120,107 +128,158 @@ export const AuthProvider: React.FC<Props> = ({ children }) => {
       setUser(null);
       setError(null);
     } catch (e) {
-      setError(e as Error);
+      setFirebaseError(e as FirebaseError | Error);
       setUser(null);
     }
   };
 
-  const signup = async (
+  /**
+   * @returns If the phone number already exists as an account.
+   */
+  const doesUserExist = async (phone: string): Promise<boolean> => {
+    try {
+      const auth = getAuth(app);
+      const signInMethods = await fetchSignInMethodsForEmail(auth, phoneNumberToEmail(phone));
+      return signInMethods.length !== 0;
+    } catch (e) {
+      setFirebaseError(e as FirebaseError | Error);
+      return false;
+    }
+  };
+
+  /**
+   * Sends an SMS code for to verify the correct phone number.
+   * @param phone The phone number (in +.... format)
+   * @param recaptcha The recaptchaVerifier set up on the screen.
+   * @returns A verificationId to be used with verifyPhone.
+   */
+  const sendSMSCode = async (
+    phone: string,
+    recaptcha: ApplicationVerifier,
+    mode: string
+  ): Promise<string> => {
+    try {
+      const auth = getAuth(app);
+
+      const signInMethods = await fetchSignInMethodsForEmail(auth, phoneNumberToEmail(phone));
+
+      // if the user is trying to reset their password but no identifier (email) exists in firebase, throw error
+      if (mode === 'reset' && signInMethods.length === 0) {
+        setError(new Error('Phone number is not registered!'));
+        return '';
+      }
```

Comment:
I think we can use doesUserExist which we defined earlier instead of rewriting some of this code